### ISSUE TITLE
App stalls/hangs 

### ISSUE THREAD

**Issue by HarrievG** [n4ze3m/page-assist#246]
Running latest edge
Version 130.0.2849.80 (Official build) (64-bit)
and latest page-assist
Version: 1 . 3 . 3

This has been an issue from the start. 
Every couple of seconds it hangs for a second. No other webpage or app/extension in the edge browser is doing this.

After inspecting the logs it seems to coincide with the /api/tags update.

2024/11/11 - 13:29:28 | 200 |      2.6771ms |       127.0.0.1 | GET      "/api/tags"
2024/11/11 - 13:29:50 | 200 |      2.2569ms |       127.0.0.1 | GET      "/api/tags"
2024/11/11 - 13:30:14 | 200 |      3.7969ms |       127.0.0.1 | GET      "/api/tags"
2024/11/11 - 13:30:38 | 200 |      2.4413ms |       127.0.0.1 | GET      "/api/tags"
2024/11/11 - 13:31:02 | 200 |      2.8498ms |       127.0.0.1 | GET      "/api/tags"
2024/11/11 - 13:31:24 | 200 |      2.8495ms |       127.0.0.1 | GET      "/api/tags"
2024/11/11 - 13:31:48 | 200 |      2.8618ms |       127.0.0.1 | GET      "/api/tags"
2024/11/11 - 13:32:12 | 200 |      2.0998ms |       127.0.0.1 | GET      "/api/tags"

Why are these polls actually needed every 23 sec? Its even polling while you do not use the dropdown or even when the browser tab is not active? 

The stall is disruptive and annoying. 

Let me know if you need any more info to pinpoint and solve this.

**Comment by HarrievG** [comment]
After update I will confirm and close bug right away. 

Thanks a lot!

**Comment by HarrievG** [comment]
This has gotten worse since I added 2 new models, it is totally unusable now. :"( 

I tried to profile it to get some impression where this is actually coming from.
In a 17 second trace made with the devtools in Edge, there is a total of 8 seconds of "Minor GC" . 

This trace was captured without ollama running, so im not sure if this is all going to be solved by your change in ollama.ts. 

When are you going to release 1.3.5? Do you have any idea what is causing this?

**Comment by n4ze3m** [comment]
Needs a few more tests before release. It's currently in draft but will be ready soon. :)

**Comment by HarrievG** [comment]
I just tried again, and unfortunately it did not help.
While cleaning up i deleted all knowledge, and suddenly it become responsive again. so it might be something in regards to that.

**Comment by n4ze3m** [comment]
Yeah, that might be the issue since we are saving files, chunks, and embeddings, which could cause problems. Also, how many knowledge bases did you have when this issue occurred?

**Comment by HarrievG** [comment]
Around 5, not super big in size, mostly pdf collections of source code. 
I keep an eye out on it when I hit it again, ill reply here.

**Comment by n4ze3m** [comment]
Version 1.3.8 is waiting for approval on the Chrome Web Store. It may take 2–3 days to get approved. Version 1.3.8 includes a fix to stop tracking fully processed files from the knowledge base, which helps reduce size and prevent hangs as the storage grows. I'll let you know once it is released on Chrome. It is currently available on Firefox
